rfcs/0017-state-snapshotting.md

```diff
@@ -0,0 +1,244 @@
+## Summary
+[summary]: #summary
+
+This RFC provides information on taking state snapshots. These snapshots would
+give us a mechanism for more quickly reproducing edge cases and errors as they
+arise. Snapshots can be broken into three projects: Partial and full-offline
+snapshots (covered here in detail), and full-online snapshots which is
+out-of-scope for now. Partial snapshotting can be done first and in parallel
+with implementation of transition frontier persistance to make the full-offline
+version much quicker to implement afterwards.
+
+## Motivation
+
+[motivation]: #motivation
+
+As stated above, snaphotting parts of our application state will give us a
+mechanism for more quickly reproducing edge cases and errors as they arive.
+
+There are several stages. Each of which provides a checkpoint that gives us
+useful features.
+
+[partial]: #partial
+### Partial Snapshotting:
+
+Snapshot enough state to be able to produce the immediate next transition, and
+then halt.
+
+Why is this useful? We've wasted a lot of time waiting for the network to reach
+a certain failing block. This feature would save us time in the future. We can
+also easily make unit tests around edge-cases we fix by saving snapshots of
+those cases.
+
+[full-offline]: #full-offline
+### Full-offline Snapshotting
+
+Snapshot enough state to recreate a full-node that can continue from the point
+of the snapshot to produce more than just one more transition.
+
+Why is this useful? Sometimes we need more context than the exact block that
+fails. This would give us that context.
```

Comment:
Can you give examples of when we need full-offline snapshotting and full-online snapshotting? For me, I think if I am producing more than one transition, I should be synchronized with the network, right?